lego-mindstorms-pieces.py

```diff
@@ -221,49 +268,67 @@ def order(shop=None, browser=None, lego_set=None, order_list=None, username=None
     browser.execute_script("window.scroll(0, 750);")
 
     print("That's gonna be crazy: {count} elements to order! Let's rock.".format(count=len(order_list)))
-    element_field = wait.until(EC.element_to_be_clickable(
-        (By.ID, 'element-filter')))
     print()
 
+    counter = 0
+    counter_outOfStock = 0
+    counter_notInSet = 0
+    counter_found = 0
+
+    total_elements=len(order_list)
+
     for brick in order_list:
         part_no, quantity = brick.split(':')
-        print("- {qty}x #{pn} ".format(qty=quantity, pn=part_no), end='')
 
+        counter += 1
+
+        print("- [{counter}/{total_elements}] {qty}x #{pn} ".format(qty=quantity, pn=part_no,counter=counter,total_elements=total_elements), end='')
+
+        element_field = wait.until(EC.element_to_be_clickable((By.ID, 'element-filter')))
         element_field.clear()
         element_field.send_keys(part_no)
         element_field.send_keys(Keys.RETURN)
         sleep(.3)  # seconds
 
         try:
             add_button = browser.find_element_by_css_selector('.element-details + button')
-            add_button.click()
-            sleep(.2)  # seconds
+            if add_button.is_enabled():
+              add_button.click()
+              sleep(.2)  # seconds
+            else:
+              counter_outOfStock += 1
+              print("NOTE: item out of stock. ")
+              continue
+
+            print("Found ", end='')
+
+            # set the value for item's quantity drop-down menu
+            amount_select = browser.find_elements_by_css_selector('.bag-item select')[-1]
+            Select(amount_select).select_by_visible_text(quantity)
+
+            # ensure the value is correct
+            selected = Select(amount_select).first_selected_option
+
+            if quantity != selected.text:
+              print("WARNING: Could not select desired quantity. {} != {}".format(quantity, selected.text))
+            else:
+              counter_found +=1
+              print()
+
         except NoSuchElementException:
             print("OOOPS! No LEGO part with that number found in set #{set}. :-(".format(set=lego_set))
+            counter_notInSet += 1
```

Comment:
No camel case, please! Should better be `not_in_set_counter` (and probably `found_counter` above).